finagle-core/src/test/scala/com/twitter/finagle/loadbalancer/RoundRobinBalancerTest.scala

```diff
@@ -0,0 +1,160 @@
+package com.twitter.finagle.loadbalancer
+
+import com.twitter.app.App
+import com.twitter.conversions.time._
+import com.twitter.finagle._
+import com.twitter.finagle.stats.{StatsReceiver, NullStatsReceiver, InMemoryStatsReceiver}
+import com.twitter.util.{Function => _, _}
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+import scala.collection.mutable
+
+private[loadbalancer] trait RoundRobinSuite {
+  // number of servers
+  val N: Int = 100
+  // number of reqs
+  val R: Int = 100000
+  // tolerated variance
+  val ε: Double = 0.0001*R
+
+  trait RRServiceFactory extends ServiceFactory[Unit, Int] {
+    def meanLoad: Double
+  }
+
+  val noBrokers = new NoBrokersAvailableException
+
+  def newBal(
+    fs: Var[Traversable[RRServiceFactory]],
+    sr: StatsReceiver = NullStatsReceiver
+  ): ServiceFactory[Unit, Int] = new RoundRobinBalancer(
+    Activity(fs.map(Activity.Ok(_))),
+    statsReceiver = sr,
+    emptyException = noBrokers,
+    maxEffort = 1
+  )
+
+  def assertEven(fs: Traversable[RRServiceFactory]) {
+    val ml = fs.head.meanLoad
+    for (f <- fs) {
+      assert(math.abs(f.meanLoad - ml) < ε,
+        "ml=%f; f.ml=%f; ε=%f".format(ml, f.meanLoad, ε))
+    }
+  }
+}
+
+@RunWith(classOf[JUnitRunner])
+class RoundRobinBalancerTest extends FunSuite with App with RoundRobinSuite {
+  case class LoadedFactory(id: Int) extends RRServiceFactory {
+    @volatile var stat: Status = Status.Open
+    var load = 0
+    var sum = 0
+    var count = 0
+
+    // This isn't quite the right notion of mean load, but it's good enough.
```

Comment:
That's a good question, I cribbed this method from P2CBalancerTest and marius wrote this comment in 1179a23b. I mean, it seems fine to me.